sechub-wrapper-prepare/src/test/java/com/mercedesbenz/sechub/wrapper/prepare/modules/skopeo/SkopeoInputValidatorTest.java

```diff
@@ -0,0 +1,208 @@
+package com.mercedesbenz.sechub.wrapper.prepare.modules.skopeo;
+
+import static com.mercedesbenz.sechub.commons.model.SecHubScanConfiguration.createFromJSON;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import com.mercedesbenz.sechub.commons.model.SecHubRemoteCredentialConfiguration;
+import com.mercedesbenz.sechub.commons.model.SecHubRemoteCredentialUserData;
+import com.mercedesbenz.sechub.commons.model.SecHubRemoteDataConfiguration;
+import com.mercedesbenz.sechub.test.TestFileWriter;
+import com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperEnvironment;
+import com.mercedesbenz.sechub.wrapper.prepare.modules.PrepareWrapperInputValidatorException;
+import com.mercedesbenz.sechub.wrapper.prepare.prepare.PrepareWrapperContext;
+
+class SkopeoInputValidatorTest {
+
+    SkopeoInputValidator validatorToTest;
+
+    TestFileWriter writer;
+
+    @BeforeEach
+    void beforeEach() {
+        writer = new TestFileWriter();
+        validatorToTest = new SkopeoInputValidator();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "ubuntu:22.04", "ubuntu", "docker://ubuntu:22.04", "docker://ubuntu", "oci:busybox_ocilayout:latest", "https://hub.docker.com",
+            "docker://docker.io/library/busybox:latest", "ubuntu@sha256:26c68657ccce2cb0a31b330cb0be2b5e108d467f641c62e13ab40cbec258c68d",
+            "ghcr.io/owner/repo:tag" })
+    void validateLocation_returns_true_for_valid_docker_urls(String location) {
+        /* execute + test */
+        assertDoesNotThrow(() -> validatorToTest.validateLocation(location));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "invalid-registry ubuntu:22.04", "docker://registry/ubuntu:invalid tag", "docker://ubuntu:tag$maliciousCode",
+            "docker://ubuntu:tag|maliciousCode", "my-registry/oci:busybox_ocilayout;latest", })
+    void validateLocation_throws_IllegalArgumentException_for_invalid_docker_urls(String location) {
+        /* execute + test */
+        assertThrows(IllegalArgumentException.class, () -> validatorToTest.validateLocation(location));
```

Comment:
As mentioned before, think about throwing a PrepareWrapperInputValidatorException in case of validation errors.